src/main/java/mate/jdbc/repository/impl/RepositoryManufacturer.java

```diff
@@ -0,0 +1,130 @@
+            insertStatement.setString(1, manufacturer.getName());
+            insertStatement.setString(2, manufacturer.getCountry());
+            insertStatement.executeUpdate();
+            ResultSet resultSet = insertStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject(1, Long.class));
+                return manufacturer;
+            } else {
+                throw new DataProcessingException(ID_GET_FAILED);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    CONNECTION_EXCEPTION + RepositoryManufacturer.class.getName(),
+                    e);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        try (Connection connection = ConnectionSupplier.getConnection();
+                PreparedStatement insertStatement = connection.prepareStatement(
+                        "SELECT * "
+                                + "FROM manufacturers "
+                                + "WHERE id = ? "
+                                + "AND is_deleted IS FALSE;")) {
+            insertStatement.setLong(1, id);
+            insertStatement.executeQuery();
+            return parseResult(insertStatement.getResultSet()).stream().findFirst();
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    CONNECTION_EXCEPTION + RepositoryManufacturer.class.getName(),
+                    e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        try (Connection connection = ConnectionSupplier.getConnection();
+                Statement getAllStatement = connection.createStatement()) {
+            ResultSet resultSet = getAllStatement.executeQuery(
+                    "SELECT * "
+                            + "FROM manufacturers "
+                            + "WHERE is_deleted IS FALSE;");
+            return parseResult(resultSet);
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    CONNECTION_EXCEPTION + RepositoryManufacturer.class.getName(),
+                    e);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> update(Manufacturer manufacturer) {
+        try (Connection connection = ConnectionSupplier.getConnection();
+                PreparedStatement insertStatement = connection.prepareStatement(
+                        "UPDATE manufacturers "
+                                + "SET " + COLUMN_NAME + " = ?, " + COLUMN_COUNTRY + " = ? "
+                                + "WHERE id = ? " + "AND is_deleted IS FALSE;")) {
+            insertStatement.setString(1, manufacturer.getName());
+            insertStatement.setString(2, manufacturer.getCountry());
+            insertStatement.setLong(3, manufacturer.getId());
+            insertStatement.executeUpdate();
+            return get(manufacturer.getId());
```

Comment:
```suggestion
            return manufacturer;
```
There is no need to execute additional get()

Reply:
there is no need to return object that were passed to method if this object wasn`t changed. Maybe i should change method return type to void?